### ISSUE TITLE
CORS error when building radar

### ISSUE THREAD

**Issue by Controlix** [thoughtworks/build-your-own-radar#263]
Whatever url I copy in the https://radar.thoughtworks.com/ link, it always results in an empty radar and when I check the browser console, I see this error:
`Access to XMLHttpRequest at 'https://docs.google.com/spreadsheets/d/18A7oDuavlh89rAmqcaXpqle8QLqIvlAkoEUxcObzuUM' from origin 'https://radar.thoughtworks.com' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

I tried with a google sheet I created myself and for which I got the public link.

I did not get any login or authentication request.

**Comment by Controlix** [comment]
After adding a Chrome extension to allow CORS, the radar still stays empty. I don't get the CORS error anymore, but I get another one:
`cb=gapi.loaded_0?le=scs:230 Uncaught {error: 'idpiframe_initialization_failed', details: 'Cookies are not enabled in current environment.'}`

**Comment by Controlix** [comment]
Indeed I do *not* try this in incognito. I tried on 2 different laptops with 3 different google accounts, a colleague tried on his laptop with another account and we always get the same result.

**Comment by Controlix** [comment]
I also tried to clone the git repo and run it locally: it gives the same issues.

**Comment by Controlix** [comment]
Another issue I have when I use my corporate account, is this error:
`Fout 400: admin_policy_enforced
Access to your account data is restricted by policies within your organisation. Please contact the administrator of your organisation for more information.`

**Comment by Controlix** [comment]
Update: I just tried on Firefox and there the radar **does show up** correctly after a few seconds.
So there is a difference in behavior between Chrome (Version 103.0.5060.53 (Official Build) (64-bit)) and Firefox (100.2 (64-bit))

**Comment by devansh-sharma-tw** [comment]
@Controlix Thanks for checking. We've tried the same sheet (https://docs.google.com/spreadsheets/d/18A7oDuavlh89rAmqcaXpqle8QLqIvlAkoEUxcObzuUM) in Chrome v103 and it is working fine. We see the CORS error as well but that does not break functionality.

We are trying to reproduce your issue. We could observe your specific cookie error message, after disabling cookies entirely at a browser level. Can you pls verify if cookies are not disabled at domain (*.thoughtworks.com) or browser level ? This can be verified by running `navigator.cookieEnabled` in the browser console while on radar.thoughtworks.com .

**Comment by Controlix** [comment]
I did check `navigator.cookiesEnabled` and it returns `true`.
But still I get the error `{error: 'idpiframe_initialization_failed', details: 'Cookies are not enabled in current environment.'}`

Chrome v103 on Ubuntu 20.04

**Comment by Controlix** [comment]
I got it working: I had to disable the setting "Block third-party cookies".
However this does not look like an acceptable solution to me. What cookies do you set? I have no issues with other websites or applications that I use with the settong "Block third-party cookies" turned on.

**Comment by devansh-sharma-tw** [comment]
Glad its working for you now!

The Google auth flow is dependent on some Google side cookies, and we have observed if cookies are disabled in general the auth flow is affected. None of the application cookies that we have as part of radar.thoughtworks.com would affect the auth flow even if they're disabled.
We're planning on updating this auth flow in the future so likely then it wont be relying on cookies.

**Comment by Controlix** [comment]
It might be worth documenting this for now, as the option to block third-party cookies seems to be the default behavior (I guess).
I look forward to see the version in which this is replaced. I will continue my investigation on usability in our team with the current version and known issue.
Thank you for the help.
